### PR TITLE
docs: add architecture docs

### PR DESCRIPTION
**Pull request by wmontwe**
This is the first iteration to document our architecture. Please have a look.

Some sections are just shallow and need further details, but are out of scope for this to timebox the documentation efforts.

Resolves [#9226](https://github.com/thunderbird/thunderbird-android/issues/9226)

Depends on #9219

Rendered:
* https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/SUMMARY.md
  * https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/GLOSSARY.md
* https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/README.md
  * https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/feature-modules.md
  * https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/legacy-module-integration.md
  * https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/module-organization.md
  * https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/module-structure.md
  * https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/ui-architecture.md
  * https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/user-flows.md 
* Contributing
  * https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/contributing/testing-guide.md

### CODE REVIEW

**Comment by wmontwe** [comment]
I updated the docs to be more specific on the legacy integration and module structure.

File: docs/GLOSSARY.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+## E
+
+### E2E (End-to-End) Encryption
+
+A system of communication where only the communicating users can read the messages.
+
+### ESMTP (Extended Simple Mail Transfer Protocol)
+
+An extension of the Simple Mail Transfer Protocol (SMTP) that adds features like authentication.
```
Comment (kewisch) [inline docs/GLOSSARY.md]: Would add a link to the relvant RFCs here

Comment (kewisch) [reply docs/GLOSSARY.md]: If we keep the glossary, can you also do this for IMAP, POP3, MIME, OAuth, etc?

File: docs/GLOSSARY.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
+# 📖 Glossary of Terms
```
Comment (kewisch) [inline docs/GLOSSARY.md]: I have mixed feelings on glossaries. In many projects they are either 1) outdated 2) too verbose and explaining terms someone involved in mobile development should simply know 3) Not detailed enough so that someone who really needs to understand will need to look up more anyway or 4) not placed close enough to the actual terms so that people will look it up on the internet anyway.

In your view, how would this glossary help contributors be more effective?

File: docs/architecture/README.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+- **🧩 Modularity**: The application is divided into distinct modules with clear responsibilities
+- **🔀 Separation of Concerns**: Each module focuses on a specific aspect of the application
+- **⬇️ Dependency Inversion**: Higher-level modules do not depend on lower-level modules directly
+- **🎯 Single Responsibility**: Each component has a single responsibility
+- **🔄 API/Implementation Separation**: Clear separation between public APIs and implementation details
+- **🧹 Clean Architecture**: Separation of UI, domain, and data layers
+- **🧪 Testability**: The architecture facilitates comprehensive testing at all levels
+- **📱 Offline-First**: The application is designed to work offline with local data storage and synchronization with remote servers
+- **🚀 Multi platform Compatibility**: The architecture is designed to support future Kotlin Multiplatform adoption
```
Comment (kewisch) [inline docs/architecture/README.md]: Killer emoji game here 🚀 😄

File: docs/architecture/README.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+
+## 📝 Architecture Decision Records
+
+The [Architecture Decision Records](adr/README.md) document the architectural decisions made during the development of the project, providing context and rationale for key technical choices.
```
Comment (kewisch) [inline docs/architecture/README.md]:
```suggestion
The [Architecture Decision Records](adr/README.md) document the architectural decisions made during the development of the project, providing context and rationale for key technical choices. Reading through these decisions will improve your contributions and ensure long-term maintainability of the project.
```

File: docs/architecture/README.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+
+## 🧩 Architectural Patterns
+
+The architecture follows several key patterns to ensure maintainability, testability, and separation of concerns:
+
+### 🔄 API/Implementation Separation
+
+Each module is split into API and implementation parts:
+
+- **📝 API**: Public interfaces, models, and contracts
```
Comment (kewisch) [inline docs/architecture/README.md:46]: Could you add a line or two here to explain who the API is serving?

File: docs/architecture/README.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+3. **🖼️ UI Layer**:
+   - 🧠 ViewModels expose UI state that reflects connectivity status
+   - 🚦 UI components display appropriate indicators for offline mode
+   - 👆 User interactions are designed to work regardless of connectivity
+
+## 💉 Dependency Injection
+
+[Koin](https://insert-koin.io/) is used for dependency injection, with modules organized by feature:
+
+The application uses [Koin](https://insert-koin.io/) for dependency injection, with modules organized by feature:
```
Comment (kewisch) [inline docs/architecture/README.md]: Duplicate sentences

File: docs/architecture/README.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+// Example Koin module for a feature
+val featureModule = module {
+    viewModel { FeatureViewModel(get()) }
+    single<FeatureRepository> { FeatureRepositoryImpl(get(), get()) }
+    single<FeatureUseCase> { FeatureUseCaseImpl(get()) }
+    single<FeatureApiClient> { FeatureApiClientImpl() }
+}
+```
+
+## 🔄 Cross-Cutting Concerns
```
Comment (kewisch) [inline docs/architecture/README.md:325]: I think we might need to build a little more detail in this section, either now or over time. All of what you write makes sense, but I'm not sure I know any better on how to apply each of these items to the project. It seems a bit vague, maybe you can provide examples that are more specific to our project?

File: docs/architecture/feature-modules.md [hunk 8]
```diff
@@ -0,0 +1,522 @@
...
+#### Subfeatures:
+
+- **API** (`feature:autodiscovery:api`): Public interfaces
+- **Autoconfig** (`feature:autodiscovery:autoconfig`): Automatic configuration
+- **Service** (`feature:autodiscovery:service`): Service implementation
+- **Demo** (`feature:autodiscovery:demo`): Demonstration implementation
+
+### 💰 Funding Module
+
+The Funding module handles in-app donations and funding options.
```
Comment (kewisch) [inline docs/architecture/feature-modules.md]:
```suggestion
The Funding module handles in-app financial contributions and funding options.
```

File: docs/architecture/feature-modules.md [hunk 9]
```diff
@@ -0,0 +1,522 @@
...
+
+#### Subfeatures:
+
+- **API** (`feature:telemetry:api`): Public interfaces
+- **Noop** (`feature:telemetry:noop`): No-operation implementation
+- **Glean** (`feature:telemetry:glean`): Mozilla Glean integration
+
+## 🔌 Extending with Additional Features
+
+The modular architecture of Thunderbird for Android allows for easy extension with additional features. Here are some examples of how the app could be extended:
```
Comment (kewisch) [inline docs/architecture/feature-modules.md]:
```suggestion
The modular architecture of Thunderbird for Android allows for easy extension with additional features. To give you an idea how the app could be extended when building a new feature, here are some theoretical examples along with their structure:
```

I want to avoid promises on a calendar feature when folks read the docs, the substantial change here "theoretical" :)

File: docs/architecture/feature-modules.md [hunk 10]
```diff
@@ -0,0 +1,522 @@
...
+├── feature:calendar:impl
+├── feature:calendar:event
+│   ├── feature:calendar:event:api
+│   └── feature:calendar:event:impl
+└── feature:calendar:sync
+    ├── feature:calendar:sync:api
+    └── feature:calendar:sync:impl
+```
+
+### ✅ Todo Feature
```
Comment (kewisch) [inline docs/architecture/feature-modules.md]: I think we should condense this to 1-2 examples. They are all very similar, so more than that doesn't make it more clear.

Comment (kewisch) [reply docs/architecture/feature-modules.md]: The next section would still work even without the structure spelled out for each feature.

File: docs/architecture/legacy-module-integration.md [hunk 11]
```diff
@@ -0,0 +1,239 @@
+# 🔙 Legacy Module Integration
+
+This document outlines how existing legacy code is integrated into the new modular architecture of the application and
+the strategy for its migration. The core principle is to isolate legacy code and provide a controlled way for
+newer modules to interact with legacy functionality without becoming directly dependent on it.
+
+> **Note:** This document should be read in conjunction with [Module Structure](module-structure.md) and [Module Organization](module-organization.md) to get a complete understanding of the modular architecture.
```
Comment (kewisch) [inline docs/architecture/legacy-module-integration.md]: You can use the fancy note blocks from https://github.com/orgs/community/discussions/16925

File: docs/architecture/legacy-module-integration.md [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+4. **Bridge Implementation**: `DefaultAccountProfileLocalDataSource` in `app-common` implements the `AccountProfileLocalDataSource` interface and serves as the bridge to legacy code.
+5. **Legacy Access**: The bridge uses `LegacyAccountWrapperManager` to access legacy account data:
+   - `LegacyAccountWrapperManager` in `core:android:account` defines the contract for legacy account access
+   - `LegacyAccountWrapper` in `core:android:account` is an immutable wrapper around the legacy `LegacyAccount` class
+6. **Data Conversion**: The bridge converts between modern `AccountProfile` objects and legacy account data via `LegacyAccountWrapper`.
+7. **Dependency Injection**: The `appCommonAccountModule` in `app-common` registers `DefaultAccountProfileLocalDataSource` as implementations of the respective interface.
+
+This multi-layered approach allows newer modules to interact with legacy account functionality through clean, modern interfaces without directly depending on legacy code. It also demonstrates how bridges can be composed, with higher-level bridges (AccountProfile) building on lower-level bridges (LegacyAccountWrapper).
+
+## Testing Considerations
```
Comment (kewisch) [inline docs/architecture/legacy-module-integration.md:132]: Some more examples of well written tests for this would be great to point to.

File: docs/architecture/module-organization.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+The App Modules (`app-thunderbird` and `app-k9mail`) contain the application-specific code, including:
+- Application entry points and initialization logic
+- Final dependency injection setup
+- Navigation configuration
+- Integration with feature modules solely for that application
+- App-specific themes and resources (strings, themes, etc.)
+
+#### 🔄 App Common Module
+
+The `app-common` module acts as the central hub for shared code between both applications. This module serves as the
```
Comment (kewisch) [inline docs/architecture/module-organization.md:114]: This section might benefit from some info on what should and should not go in here, potentially with examples. For a new contributor it might be hard to decide if something is "common" code vs something that needs to be a feature module.

**Review by kewisch — APPROVED** [review]
Excellent work, this gives us a lot more context on the project. I have a few comments for your consideration. Even without thought I'd be ok merging this.

**Comment by kewisch** [comment]
@shamim-emon @marcRDZ @ClemRiviere I'm curious on your thoughts on the documentation here, does this help create a better understanding for the project? What is one thing you would improve to make it more clear? I've added links to the rendered versions of these changes in the original comment from Wolf up top.

File: docs/architecture/README.md [hunk 14]
```diff
@@ -0,0 +1,324 @@
...
+The application is organized into several module types:
+
+- **📱 App Modules**: `app-thunderbird` and `app-k9mail` - Application entry points
+- **🔄 App Common**: `app-common` - Shared code between applications
+- **✨ Feature Modules**: `feature:*` - Independent feature modules
+- **🧰 Core Modules**: `core:*` - Foundational components and utilities used across multiple features
+- **📚 Library Modules**: `library:*` - Specific implementations for reuse
+- **🔙 Legacy Modules**: Legacy code being gradually migrated
+
+For more details on the module structure, see the [Module Structure](module-structure.md) document.
```
Comment (wmontwe) [inline docs/architecture/README.md]: This should link to the new module organization

File: docs/architecture/legacy-module-integration.md [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+    class APP_COMMON app_common
+    class OTHER_MODULES,INTERFACES,IMPLEMENTATIONS new_modules
+    class LEGACY1,LEGACY2,LEGACY3 legacy
+```
+
+### Implementation Techniques
+
+Several techniques are used to implement the bridge pattern effectively:
+
+1. **Wrapper Classes**: Creating immutable data classes that wrap legacy data structures, implementing interfaces from the new architecture. These wrappers typically include conversion methods to transform between legacy and new data structures.
```
Comment (wmontwe) [inline docs/architecture/legacy-module-integration.md]: The wrappers should not contain conversion methods. They should delegate this to specific mapper classes.

File: docs/architecture/legacy-module-integration.md [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+
+2. **Adapter Implementations**: Classes in `:app-common` that implement interfaces from the new architecture but delegate to legacy code internally.
+
+3. **Data Conversion**: Static conversion methods (often in companion objects) that handle mapping between legacy and new data structures, ensuring clean separation of concerns.
```
Comment (wmontwe) [inline docs/architecture/legacy-module-integration.md]: This should be handled by dedicated mappers

File: docs/architecture/legacy-module-integration.md [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+7. **Dependency Injection**: The `appCommonAccountModule` in `app-common` registers `DefaultAccountProfileLocalDataSource` as implementations of the respective interface.
+
+This multi-layered approach allows newer modules to interact with legacy account functionality through clean, modern interfaces without directly depending on legacy code. It also demonstrates how bridges can be composed, with higher-level bridges (AccountProfile) building on lower-level bridges (LegacyAccountWrapper).
+
+## Testing Considerations
+
+Testing bridge implementations requires special attention to ensure both the bridge itself and its integration with legacy code work correctly:
+
+1. **Unit Testing Bridge Classes**:
+   - Test the bridge implementation in isolation by mocking/stubbing the legacy dependencies
```
Comment (wmontwe) [inline docs/architecture/legacy-module-integration.md]: Remove mocking

File: docs/architecture/legacy-module-integration.md (hunk 12 above)
Comment (wmontwe) [reply docs/architecture/legacy-module-integration.md:132]: This is the hard part, hence there are no good examples. It mostly depends on which legacy component you're trying to wrap. So they are tailor fit for the problem at hand and might even skip testing some aspects if the involved legacy code is not testable (meaning high effort and time consuming).

File: docs/architecture/legacy-module-integration.md [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
+
+### Migration Example
+
+Using the account profile example, the migration process would look like:
+
+1. **Identify**: Account profile functionality in legacy modules needs modernization.
+2. **Define Interfaces**:
+   - `AccountProfileRepository` interface is defined in `feature:account:api`
+   - `AccountProfileLocalDataSource` interface is defined in `feature:account:core`
+3. **Implement**: Create a new implementation of `AccountProfileLocalDataSource` in a modern module, e.g., `feature:account:impl`.
```
Comment (wmontwe) [inline docs/architecture/legacy-module-integration.md]: We need to add entity modeling here too as interfaces are not enough.

File: docs/architecture/legacy-module-integration.md [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
+4. **Update Bridge**: Update or remove `DefaultAccountProfileLocalDataSource` in `app-common`.
+5. **Switch DI**: Update `appCommonAccountModule` to provide the new implementation instead of `CommonAccountProfileLocalDataSource`.
```
Comment (wmontwe) [inline docs/architecture/legacy-module-integration.md]: DefaultAccountProfileLocalDataSource

File: docs/architecture/legacy-module-integration.md [hunk 20]
```diff
@@ -0,0 +1,239 @@
...
+6. **Retire**: Once all references to legacy account code are removed, the legacy code and lower-level bridges (`LegacyAccountWrapperManager`, `CommonLegacyAccountWrapperManager`) can be safely deleted.
```
Comment (wmontwe) [inline docs/architecture/legacy-module-integration.md]: DefaultLegacyAccountWrapperManager

File: docs/architecture/module-structure.md [hunk 21]
```diff
@@ -0,0 +1,402 @@
...
+
+API modules should follow the naming convention:
+- `feature:<feature-name>:api` for feature modules
+- `core:<core-name>:api` for core modules
+
+#### Example structure for a feature API module:
+
+```bash
+feature:account:api
+├── src/main/kotlin/app/k9mail/feature/account/api
```
Comment (wmontwe) [inline docs/architecture/module-structure.md]: use thunderbird here

File: docs/architecture/module-structure.md [hunk 22]
```diff
@@ -0,0 +1,402 @@
...
+
+- **Reusability**: Create reusable test utilities and data factories
+- **Isolation**: Tests should be isolated and not depend on external systems
+- **Readability**: Tests should be easy to read and understand
+- **Maintainability**: Tests should be easy to maintain and update
+- **Coverage**: Tests should cover all critical paths and edge cases
+
+### 🤖 Fake Module
+
+Fake modules provide alternative implementations of interfaces for testing, development, or demonstration purposes. They are simpler than the real implementations and are designed to be used in controlled environments.
```
Comment (wmontwe) [inline docs/architecture/module-structure.md:170]: Limit this to the most generic data, specific use cases or test setups should be part of the actual test.

File: docs/architecture/module-structure.md [hunk 23]
```diff
@@ -0,0 +1,402 @@
...
+- **Dependency Visibility**: Use appropriate visibility modifiers to limit access to implementation details
+
+### Dependency Injection
+
+- Use Koin for dependency injection
+- Configure module dependencies in dedicated Koin modules
+- Inject API interfaces, not implementation classes
+- Use lazy injection where appropriate to improve startup performance
+
+```mermaid
```
Comment (wmontwe) [inline docs/architecture/module-structure.md:275]: The diagram needs to be moved up

File: docs/GLOSSARY.md (hunk 2 above)
Comment (wmontwe) [reply docs/GLOSSARY.md]: I think we could remove the glossary, it's not really needed.

File: docs/SUMMARY.md [hunk 24]
```diff
@@ -6,25 +6,32 @@
...
-- [Design](DESIGN.md)
+- [Architecture](architecture/README.md)
+  - [Module Organization](architecture/module-organization.md)
+  - [Module Structure](architecture/module-structure.md)
+  - [Feature Modules](architecture/feature-modules.md)
+  - [UI Architecture](architecture/ui-architecture.md)
+  - [User Flows](architecture/user-flows.md)
+  - [Legacy Module Integration](architecture/legacy-module-integration.md)
+  - [Architecture Decision Records](architecture/adr/README.md)
+    - [Accepted]()
```
Comment (rafaeltonholo) [inline docs/SUMMARY.md:22]: If the reader clicks on the `Accepted` link, it will redirect to the `docs/` folder. Is this the intention?

File: docs/SUMMARY.md [hunk 25]
```diff
@@ -6,25 +6,32 @@
...
+      - [0001 - Switch From Java to Kotlin](architecture/adr/0001-switch-from-java-to-kotlin.md)
+      - [0002 - UI - Wrap Material Components in Atomic Design System](architecture/adr/0002-ui-wrap-material-components-in-atomic-design-system.md)
+      - [0003 - Test - Switch Test Assertions From Truth to Assertk](architecture/adr/0003-switch-test-assertions-from-truth-to-assertk.md)
+      - [0004 - Naming Conventions for Interfaces and Their Implementations](architecture/adr/0004-naming-conventions-for-interfaces-and-their-implementations.md)
+      - [0005 - Central Project Configuration](architecture/adr/0005-central-project-configuration.md)
+      - [0006 - White Label Architecture](architecture/adr/0006-white-label-architecture.md)
+      - [0007 - Project Structure](architecture/adr/0007-project-structure.md)
+      - [0008 - Change Shared Module package to `net.thunderbird`](architecture/adr/0008-change-shared-modules-package-name.md)
+    - [Proposed]()
```
Comment (rafaeltonholo) [inline docs/SUMMARY.md:31]: If the reader clicks on the `Proposed` link, it will redirect to the `docs/` folder. Is this the intention?

File: docs/SUMMARY.md [hunk 26]
```diff
@@ -6,25 +6,32 @@
...
+    - [Rejected]()
```
Comment (rafaeltonholo) [inline docs/SUMMARY.md:32]: If the reader clicks on the `Rejected` link, it will redirect to the `docs/` folder. Is this the intention?

File: docs/architecture/README.md (hunk 5 above)
Comment (rafaeltonholo) [reply docs/architecture/README.md:46]: I would also consider adding an example of both modules where contributors could look at and base on. It could be just a link to the module

File: docs/architecture/README.md [hunk 27]
```diff
@@ -0,0 +1,324 @@
...
+    UI --> VM
+    VM --> UC
+    UC --> REPO
+    REPO --> DS
+    DS --> API
+    DS --> DB
+
+    classDef uiLayer fill:#d0e0ff,stroke:#0066cc
+    classDef businessLayer fill:#d5f5d5,stroke:#00aa00
+    classDef dataLayer fill:#ffe0d0,stroke:#cc6600
```
Comment (rafaeltonholo) [inline docs/architecture/README.md]: It is a bit hard to read the text when on dark mode:
| light | dark |
| ----- | ----- |
| image | image |

File: docs/architecture/README.md [hunk 28]
```diff
@@ -0,0 +1,324 @@
...
+    classDef ucClass fill:#ffe0d0,stroke:#cc6600
+
+    class V viewClass
+    class VM vmClass
+    class UC ucClass
+```
+
+Key components:
+- **👁️ View**: Renders the UI based on the current state and sends user events to the ViewModel
+- **🧠 ViewModel**: Processes events, updates state, and triggers actions
```
Comment (rafaeltonholo) [inline docs/architecture/README.md]: I would like to propose a more detailed description for the ViewModel:

> Processes user events, converting them into actions and sending them to the Domain Layer. It also maps the results to a state and sends state updates to the UI.

File: docs/architecture/README.md [hunk 29]
```diff
@@ -0,0 +1,324 @@
...
+- **📊 State**: Immutable representation of the UI state
```
Comment (rafaeltonholo) [inline docs/architecture/README.md]: Maybe it would be interesting explaining what `Immutable` stands for and why it needs to be Immutable

File: docs/architecture/README.md [hunk 30]
```diff
@@ -0,0 +1,324 @@
...
+
+**Pattern: Data Source Pattern**
+- 🔍 Abstracts data sources behind a clean API
+-  Maps data between domain models and data transfer objects
+
+```mermaid
+graph TD
+    subgraph DOMAIN[Domain Layer]
+        REPOSITORIES[Repository]
+        M
```
Comment (rafaeltonholo) [inline docs/architecture/README.md]: Is this supposed to be Model or it is just a typo?

File: docs/architecture/README.md [hunk 31]
```diff
@@ -0,0 +1,324 @@
...
+
+### 🔒 Security
+
+- 🔐 Data encryption for sensitive information
+- 🔑 Secure authentication mechanisms
+- 🛡️ Network security with proper certificate validation
+
+## 🧪 Testing Strategy
+
+The architecture supports comprehensive testing:
```
Comment (rafaeltonholo) [inline docs/architecture/README.md:637]: I would add a link to our testing guide in this section

File: docs/architecture/feature-modules.md [hunk 32]
```diff
@@ -0,0 +1,522 @@
...
+3. **Minimal Dependencies**: Minimize dependencies between feature modules
+4. **Proper Layering**: Follow Clean Architecture principles within each feature
+5. **Testability**: Design features to be easily testable in isolation
+6. **Documentation**: Document the purpose and usage of each feature module
+7. **Consistent Naming**: Follow the established naming conventions
+8. **Feature Flags**: Use feature flags for gradual rollout and A/B testing
+9. **Accessibility**: Ensure all features are accessible to all users
+10. **Internationalization**: Design features with internationalization in mind
+
+By following these guidelines, the Thunderbird for Android application can maintain a clean, modular architecture while expanding its functionality to meet user needs.
```
Comment (rafaeltonholo) [inline docs/architecture/feature-modules.md]: I would move this entire section to the top of the document. I think this information would be searched for more often than the others.

File: docs/architecture/README.md (hunk 3 above)
Comment (marcRDZ) [inline docs/architecture/README.md]: IMHO these last two items deserve being at first position, they are specific decisions on the project together with the modularity approach while the rest are more generically speaking SOLID principles and CLEAN pattern architecture.

**Comment by marcRDZ** [comment]
Nice work! Not only to make an easier landing on contributions but also to show how to work and take decisions on such huge projects.
I'd like to review it deeper.

File: docs/GLOSSARY.md (hunk 1 above)
Comment (wmontwe) [reply docs/GLOSSARY.md]: I removed the glossary.

File: docs/GLOSSARY.md (hunk 2 above)
Comment (wmontwe) [reply docs/GLOSSARY.md]: I removed the glossary.

File: docs/architecture/legacy-module-integration.md (hunk 11 above)
Comment (wmontwe) [reply docs/architecture/legacy-module-integration.md]: There is support for mdbook, added it.

File: docs/SUMMARY.md (hunk 25 above)
Comment (wmontwe) [reply docs/SUMMARY.md:31]: This file is used for [mdbook](https://rust-lang.github.io/mdBook/) to generate the navigation and documentation content.

I added a disclaimer to the file.

File: docs/SUMMARY.md (hunk 26 above)
Comment (wmontwe) [reply docs/SUMMARY.md:32]: This file is used for [mdbook](https://rust-lang.github.io/mdBook/) to generate the navigation and documentation content.

I added a disclaimer to the file.

File: docs/SUMMARY.md (hunk 24 above)
Comment (wmontwe) [reply docs/SUMMARY.md:22]: This file is used for [mdbook](https://rust-lang.github.io/mdBook/) to generate the navigation and documentation content.

I added a disclaimer to the file.

File: docs/architecture/README.md [hunk 33]
```diff
@@ -0,0 +1,325 @@
...
+    single<FeatureUseCase> { FeatureUseCaseImpl(get()) }
+    single<FeatureApiClient> { FeatureApiClientImpl() }
+}
+```
+
+## 🔄 Cross-Cutting Concerns
+
+### ⚠️ Error Handling
+
+- 🧠 Domain errors for business logic errors
```
Comment (marcRDZ) [inline docs/architecture/README.md]: May be some examples of types to have a clear sight on the different scopes of error handling could be helpful.

File: docs/architecture/README.md (hunk 28 above)
Comment (marcRDZ) [reply docs/architecture/README.md]: Adding the distinction between states and effects with underlying links to corresponding sections could be cool!
[States](https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/ui-architecture.md#-state), [Events](https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/ui-architecture.md#-events), [Effects](https://github.com/wmontwe/thunderbird-android/blob/add-architecture-docs/docs/architecture/ui-architecture.md#effects)

File: docs/architecture/legacy-module-integration.md [hunk 34]
```diff
@@ -0,0 +1,240 @@
...
+   - Test the bridge with actual legacy code to ensure proper integration
+   - Verify that the bridge correctly handles all edge cases from legacy code
+3. **Test Doubles**:
+   - Create fake implementations of bridge classes for testing other components
+   - Example: `FakeLegacyAccountWrapperManager` can be used to test components that depend on `LegacyAccountWrapperManager`
+4. **Migration Testing**:
+   - When migrating from a legacy bridge to a new implementation, test both implementations with the same test suite
+   - Ensure behavior consistency during the transition
+
+## Migration Strategy
```
Comment (marcRDZ) [inline docs/architecture/legacy-module-integration.md:344]: Great guide! I would love to tackle some issue regarding migration of features to follow it ;)

File: docs/architecture/user-flows.md [hunk 35]
```diff
@@ -0,0 +1,15 @@
+# User Flows
```
Comment (marcRDZ) [inline docs/architecture/user-flows.md:1]: Really cool!

File: docs/architecture/ui-architecture.md [hunk 36]
```diff
@@ -0,0 +1,314 @@
...
+    data object NavigateNext : AccountSettingsEffect
+    data object NavigateBack : AccountSettingsEffect
+}
+```
+
+### Effects
+
+- Represent side effects or navigation actions
+- Emitted by the ViewModel to trigger navigation or other actions
+- Handled by the UI layer to perform navigation or show messages
```
Comment (rafaeltonholo) [inline docs/architecture/ui-architecture.md]: I would include an example of effect in this section as well

File: docs/architecture/ui-architecture.md [hunk 37]
```diff
@@ -0,0 +1,314 @@
...
+- **🔙 Back Stack Management**: Handles the navigation back stack
+- **↩️ Deep Linking**: Supports deep linking to specific screens
+
+TODO: explain how to set up navigation in the app, including the navigation graph and how to navigate between screens.
+
+## 🎭 Theming and Customization
+
+The UI architecture supports comprehensive theming and customization:
+
+(colors, elevations, images, shapes, sizes, spacings, typography)
```
Comment (rafaeltonholo) [inline docs/architecture/ui-architecture.md]: I guess this line is redundant and it is already explained below

File: docs/architecture/ui-architecture.md [hunk 38]
```diff
@@ -0,0 +1,314 @@
...
+                    onEvent(AccountSettingsEvent.SettingChanged(setting, value))
+                },
+                onSaveClicked = { onEvent(AccountSettingsEvent.SaveClicked) }
+            )
+        }
+    }
+}
+```
+
+### 🎨 Design System Components
```
Comment (rafaeltonholo) [inline docs/architecture/ui-architecture.md:108]: I believe it would be good to mention that we use the [Atomic Design Methodology](https://atomicdesign.bradfrost.com/chapter-2/) and add a brief explanation of the difference between an [atom](https://atomicdesign.bradfrost.com/chapter-2/#atoms), a [molecule](https://atomicdesign.bradfrost.com/chapter-2/#molecules), an [organism](https://atomicdesign.bradfrost.com/chapter-2/#organisms), and a [template](https://atomicdesign.bradfrost.com/chapter-2/#templates) when creating a new  Design System component. Perhaps just a link to the Atomic Design Methodology would be enough, not sure.

File: docs/architecture/ui-architecture.md [hunk 39]
```diff
@@ -0,0 +1,314 @@
...
+  - **🌓 Dark Mode**: Full support for light and dark themes
+  - **🌈 Dynamic Color**: Support for dynamic color based on system settings
+- **🪜 Elevations**: Consistent elevation system for shadows
+- **🖼️ Images**: Images and icons consistent with the theme
+- **🔶 Shapes**: Customizable shape system for components
+- **📐 Sizes**: Standardized sizes for components
+- **📏 Spacings**: Consistent spacing system for layout
+- **🅰️ Typography**: Consistent typography system
+
+## ♿ Accessibility
```
Comment (rafaeltonholo) [inline docs/architecture/ui-architecture.md:1293]: Not related to this documentation, but I would like to eventually have a dedicated Accessibility page as this is a very important matter. The page would explain our approach to accessibility, outline our targets and goals, and provide guidance on implementation.

File: docs/architecture/user-flows.md (hunk 35 above)
Comment (rafaeltonholo) [inline docs/architecture/user-flows.md:1]: Would it be worth mentioning that we have (or plan to have) maestro tests covering the user flows?

File: docs/architecture/README.md (hunk 5 above)
Comment (wmontwe) [reply docs/architecture/README.md:46]: I added a link to the more detailed explanation in the module-structure document.

File: docs/architecture/README.md (hunk 7 above)
Comment (wmontwe) [reply docs/architecture/README.md:325]: I added more details.

File: docs/architecture/README.md (hunk 27 above)
Comment (wmontwe) [reply docs/architecture/README.md]: I didn't test with dark mode, need to adjust the text color too.

File: docs/architecture/README.md (hunk 30 above)
Comment (wmontwe) [reply docs/architecture/README.md]: Leftover from a previous version.

File: docs/architecture/ui-architecture.md (hunk 39 above)
Comment (wmontwe) [reply docs/architecture/ui-architecture.md:1293]: That would be great, but I would defer that to later.

File: docs/architecture/README.md (hunk 33 above)
Comment (wmontwe) [reply docs/architecture/README.md]: Added some examples.

File: docs/architecture/ui-architecture.md (hunk 38 above)
Comment (wmontwe) [reply docs/architecture/ui-architecture.md:108]: Yes, but the terminology changed a bit and I would update this in a separate task, especially as the new design system will define a lot of the foundation.

File: docs/architecture/legacy-module-integration.md (hunk 34 above)
Comment (wmontwe) [reply docs/architecture/legacy-module-integration.md:344]: @marcRDZ We don't have a detailed plan yet, but in the meantime, you could have a look at the legacy code base and propose changes to work on.

File: docs/architecture/ui-architecture.md (hunk 36 above)
Comment (wmontwe) [reply docs/architecture/ui-architecture.md]: The section is now extended with more details and examples.
